builder/dockerfile/shell_parser.go

```diff
@@ -286,10 +296,10 @@ func (sw *shellWord) processDollar() (string, error) {
 			return newValue, nil
 
 		default:
-			return "", errors.Errorf("unsupported modifier (%c) in substitution: %s", modifier, sw.word)
+			return "", errors.Errorf("unsupported modifier (%c) in substitution", modifier)
 		}
 	}
-	return "", errors.Errorf("missing ':' in substitution: %s", sw.word)
+	return "", errors.Errorf("missing ':' in substitution")
```

Comment:
The main reason is just trying to reduce the number of fields on the struct.

The context is still shown to users with the change I made to `process()` (it wraps the error). With this change all errors will include the input.